src/typescript/frontend/src/utils/geolocation.ts

```diff
@@ -0,0 +1,44 @@
+import { LazyPromise } from "@sdk/utils";
+import { IPINFO_TOKEN } from "lib/env";
+
+const COUNTRY_CACHING_DURATION = 1000 * 60 * 60 * 24; // One day
+const COUNTRY_LOCALSTORAGE_KEY = "user-country";
+
+type LocationCache = {
+  country: string;
+  expiry: number;
+};
+
+const isAllowedCountry = (country: string) => {
+  return country !== "US" && country !== "KP" && country !== "IR";
+};
+
+const getCountry: () => Promise<string> = async () => {
+  const cache = localStorage.getItem(COUNTRY_LOCALSTORAGE_KEY);
+
+  if (cache) {
+    const data: LocationCache = JSON.parse(cache);
+    if (data.expiry > new Date().getTime()) {
+      return data.country;
+    }
+  }
+
+  const queryResult = await fetch(`https://ipinfo.io/json?token=${IPINFO_TOKEN}`)
```

Comment:
If we accidentally ddos this or site goes down no one can get into the site, yea? Right now I can only get to the verify screen btw, so no one will be able to see anything if that happens